### PR TITLE
feat(email-validation): change email validation to be rfc2822 compliant

### PR DESCRIPTION
**Pull request by haidarCheaito**
This PR is made to make email validation on keycloak side RFC compliant, to avoid errors while sending emails.

### CODE REVIEW

**Review by windwinkel — APPROVED** [review]

File: server-spi-private/src/main/java/org/keycloak/validate/validators/EmailValidator.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 import java.util.Collections;
 import java.util.List;
-import java.util.regex.Pattern;
+
```
Comment (dniggemann) [inline server-spi-private/src/main/java/org/keycloak/validate/validators/EmailValidator.java:21]: Remove empty line?

File: services/pom.xml [hunk 2]
```diff
@@ -266,6 +266,7 @@
 
                         <!-- Replace the dependencies that aren't in Maven Central -->
                         <dependencies>
+
```
Comment (dniggemann) [inline services/pom.xml]: Remove empty line?

**Review by dniggemann — APPROVED** [review]
One minor comment: We don't have an auto code formatter for java and xml files, so I commented 2 lines, that look strange in the diff.

Rest looks good. Approved already.

**Comment by zyv** [comment]
Not documented in the README, @pchorus @windwinkel ?

**Comment by mm-matthias** [comment]
@windwinkel Are there any plans to upstream this to Keycloak itself?

**Pull request merged**